presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedBucketNodeMap.java

```diff
@@ -41,7 +41,10 @@ public FixedBucketNodeMap(ToIntFunction<Split> splitToBucket, List<InternalNode>
     @Override
     public Optional<InternalNode> getAssignedNode(int bucketedId)
     {
-        return Optional.of(bucketToNode.get(bucketedId));
+        if (bucketToNode.size() > bucketedId) {
+            return Optional.of(bucketToNode.get(bucketedId));
+        }
+        return Optional.empty();
```

Comment:
When will this happen?

Reply:
It seems to be appearing in some flaky tests which I haven't been able to reproduce, was trying to add proper handling so that error can surface out with more debug info

Reply:
@highker Here is a stack trace: #14797

Reply:
@kewang1024, if you really wanna expose the error message, just throw it here and add a TODO